wagtail/wagtailusers/forms.py

```diff
@@ -233,19 +239,44 @@ def save(self):
         return group
 
 
-class GroupPagePermissionForm(forms.ModelForm):
-    page = forms.ModelChoiceField(queryset=Page.objects.all(),
-                                  widget=AdminPageChooser(show_edit_link=False, can_choose_root=True))
+class PagePermissionsForm(forms.Form):
+    """
+    Note 'Permissions' (plural). A single instance of this form defines the permissions
+    that are assigned to an entity (i.e. group or user) for a specific page.
+    """
+    page = forms.ModelChoiceField(
+        queryset=Page.objects.all(),
+        widget=AdminPageChooser(show_edit_link=False, can_choose_root=True)
+    )
+    permission_types = forms.MultipleChoiceField(
+        choices=PAGE_PERMISSION_TYPE_CHOICES,
+        required=False,
+        widget=forms.CheckboxSelectMultiple
+    )
 
-    class Meta:
-        model = GroupPagePermission
-        fields = ('page', 'permission_type')
 
+class BaseGroupPagePermissionFormSet(forms.BaseFormSet):
```

Comment:
It could probably be done, but the two classes don't quite correspond (mainly because `GroupPagePermission` uses a 'choices' list for the permission type, rather than a real foreign key to auth.Permission), and making things fully generic has been a bit of a pain point for this PR - so I'd prefer to leave this for a future task :-)